hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -180,33 +169,25 @@ public boolean hasNext() {
   public T next() {
     checkSanityOfProcess();
     currentKey = nextKey;
-    nextKey = null;
-    boolean keyFound = false;
-    while (!keyFound) {
-      try {
-        Optional<Integer> keyLength = getNextNumberInStream();
-        if (!keyLength.isPresent()) {
-          return getTransformedValue(currentKey);
-        }
-        String keyStr = readNextNumberOfCharsFromStream(keyLength.get());
-        Matcher matcher = PATTERN_MATCHER.matcher(keyStr);
-        if (keyStr.length() == keyLength.get() && matcher.find()) {
-          Optional<Integer> valueLength = getNextNumberInStream();
-          if (valueLength.isPresent()) {
-            String valueStr = readNextNumberOfCharsFromStream(
-                valueLength.get());
-            if (valueStr.length() == valueLength.get()) {
-              keyFound = true;
-              nextKey = new KeyValue(matcher.group(PATTERN_KEY_GROUP_NUMBER),
-                  matcher.group(PATTERN_SEQ_GROUP_NUMBER),
-                  matcher.group(PATTERN_TYPE_GROUP_NUMBER),
-                  valueStr);
+    nextKey = Optional.empty();
+    try {
```

Comment:
nit: next() has too much nesting and could be simplified to this.
```suggestion
    try {
      Optional<byte[]> key = getNextByteArray();
      if (!key.isPresent()) {
        return getTransformedValue(currentKey);
      }

      Optional<UnsignedLong> sequenceNumber = getNextUnsignedLong();
      if (!sequenceNumber.isPresent()) {
        return getTransformedValue(currentKey);
      }

      Optional<Integer> type = getNextNumberInStream();
      if (!type.isPresent()) {
        return getTransformedValue(currentKey);
      }

      Optional<byte[]> value = getNextByteArray();
      value.ifPresent(bytes -> nextKey = Optional.of(
          new KeyValue(key.get(), sequenceNumber.get(), type.get(), bytes)));
      return getTransformedValue(currentKey);
    } catch (IOException e) {
      throw new RuntimeIOException(e);
    }
```

Reply:
done